sdk/core/src/services/AffinidiVaultStorageService.ts

```diff
@@ -0,0 +1,184 @@
+
+    for (const credential of credentials) {
+      let credentialId = credential?.id
+      if (!isW3cCredential(credential)) {
+        credentialId = credential?.data?.id
+      }
+
+      const credentialIdHash = await this._platformEncryptionTools.computePersonalHash(privateKeyBuffer, credentialId)
+
+      const typeHashes = []
+      if (isW3cCredential(credential)) {
+        for (const credentialType of credential.type) {
+          const typeHash = await this._platformEncryptionTools.computePersonalHash(privateKeyBuffer, credentialType)
+          typeHashes.push(typeHash)
+        }
+      }
+
+      const cyphertext = await this._platformEncryptionTools.encryptByPublicKey(publicKeyBuffer, credential)
+
+      encryptedCredentials.push({
+        credentialId: credentialIdHash,
+        credentialTypes: typeHashes,
+        payload: cyphertext,
+      })
+    }
+
+    return encryptedCredentials
+  }
+
+  private async _computeTypesHashes(types: string[][]): Promise<string[][]> {
+    const privateKeyBuffer = this._keysService.getOwnPrivateKey()
+
+    const hashedTypes: string[][] = []
+    for (const subset of types) {
+      const hashedSubset: string[] = []
+      for (const type of subset) {
+        const hashedType = await this._platformEncryptionTools.computePersonalHash(privateKeyBuffer, type)
+        hashedSubset.push(hashedType)
+      }
+
+      hashedTypes.push(hashedSubset)
+    }
+
+    return hashedTypes
+  }
+
+  private async _decryptCredentials(encryptedCredentials: VaultCredential[]): Promise<any[]> {
+    const credentials: any[] = []
+
+    for (const encryptedCredential of encryptedCredentials) {
+      const credential = await this._decryptCredential(encryptedCredential)
+      credentials.push(credential)
+    }
+
+    return credentials
+  }
+
+  private async _decryptCredential(encryptedCredential: VaultCredential): Promise<any> {
+    const privateKeyBuffer = this._keysService.getOwnPrivateKey()
+
+    const credential = await this._platformEncryptionTools.decryptByPrivateKey(
+      privateKeyBuffer,
+      encryptedCredential.payload,
+    )
+
+    return credential
+  }
+
+  public async saveCredentials(credentials: any[], storageRegion: string): Promise<VaultCredential[]> {
+    const token = await this._authorizeVcVault()
+
+    const responses = []
+
+    const encryptedCredentials = await this._encryptCredentials(credentials)
+
+    for (const credential of encryptedCredentials) {
+      const { body } = await this._vaultApiService.storeCredential(token, storageRegion, credential.credentialId, {
+        credentialTypes: credential.credentialTypes,
+        payload: credential.payload,
+      })
+
+      responses.push(body)
+    }
+
+    return responses
+  }
+
+  public async searchCredentials(types: string[][], storageRegion: string): Promise<any[]> {
+    const token = await this._authorizeVcVault()
+
+    const hashedTypes = await this._computeTypesHashes(types)
```

Comment:
This operation is probably pretty safe, but looking at this pattern; I think we can create some class in internal api package that handles did-auth. And that helper class can do did auth before each request. For example;

```
export default class AffinidiVaultApiService extends DidAuthApiService<ApiSpec> {
  someMethodWithDidAuth() {
    const token = this.getToken();
    return this.httpRequest(...) 
  }

  someMethod() {
    return this.httpRequest(...)
   }
}

// Here we have the token impl with expiration date check
export default abstract class DidAuthService<T> extends GenericApiService<T> 
```

Reply:
Yes, something like that would be great.